tests/test_http.py

```diff
@@ -113,13 +112,12 @@ def do_POST(self):
         try:
             length = int(self.headers['Content-Length'])
             entity = io.BytesIO(self.rfile.read(length))
-            data = json.loads(gzip.GzipFile(fileobj=entity).read())
+            data = json.loads(gzip.GzipFile(fileobj=entity).read().decode())
```

Comment:
Why did it work in 3.6? Should this be specifying the charset to decode it into?

Reply:
* Python 2.7
    * [gzip read returns str](https://github.com/python/cpython/blob/2.7/Lib/gzip.py#L247-L278)
    * [json loads deserializes a str or unicode instance](https://github.com/python/cpython/blob/2.7/Lib/json/__init__.py#L294-L352)
    * str decode returns unicode
* Python 3.5
    * [gzip read returns bytes](https://github.com/python/cpython/blob/3.5/Lib/gzip.py#L269-L274)
    * [json loads deserializes a str instance](https://github.com/python/cpython/blob/3.5/Lib/json/__init__.py#L271-L332)
    * bytes decode returns a str
* Python 3.6
    * [gzip read returns bytes](https://github.com/python/cpython/blob/3.6/Lib/gzip.py#L271-L276)
    * [json loads deserializes a str, bytes or bytearray instance](https://github.com/python/cpython/blob/3.6/Lib/json/__init__.py#L302-L367)
    * bytes decode returns a str